.github/workflows/codeql-analysis.yml

```diff
@@ -0,0 +1,46 @@
+name: "CodeQL"
+
+on:
+  pull_request:
+    paths: ["**.go"]
+    branches:
+      - main
+      - release/**
+  push:
+    branches:
+      - main
+      - release/**
+    paths: ["**.go"]
+
+jobs:
+  analyze:
+    name: Analyze
+    runs-on: ubuntu-latest
+    permissions:
+      actions: read
+      contents: read
+      security-events: write
+
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.18
+      # Initializes the CodeQL tools for scanning.
+      - name: Initialize CodeQL
+        uses: github/codeql-action/init@v2
+        with:
+          languages: "go"
+          queries: crypto-com/cosmos-sdk-codeql@main,security-and-quality
+          # If you wish to specify custom queries, you can do so here or in a config file.
+          # By default, queries listed here will override any specified in a config file.
+          # Prefix the list here with "+" to use these queries and those in the config file.
+          # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+
+      # ℹ️ Command-line programs to run using the OS shell.
+      # 📚 https://git.io/JvXDl
+
+      - name: Perform CodeQL Analysis
```

Comment:
Shouldn't we compile the code before performing CodeQL Analysis? 
Looking at CodeQL readme seems like is recommended

Reply:
No, it's only needed for Go, and nice to have to avoid other complex checks which are run here. 
We compile in other workflow using different platforms (mac, linux, ...)